samples/excel/85-preview-apis/direct-precedents.yaml

```diff
@@ -0,0 +1,154 @@
+            const sheet1 = context.workbook.worksheets.add("Sample2019Data");
+            const data1 = [
+              ["Product", "Qty", "Unit Price", "Total Price 2019"],
+              ["Almonds", 2, 7.5, "=C3 * D3"],
+              ["Coffee", 1, 34.5, "=C4 * D4"],
+              ["Chocolate", 5, 9.56, "=C5 * D5"]
+            ];
+
+            const range1 = sheet1.getRange("B2:E5");
+            range1.values = data1;
+            range1.format.autofitColumns();
+
+            // Set up the second sample table.
+            const sheet2 = context.workbook.worksheets.add("Sample2020Data");
+            const data2 = [
+              ["Product", "Qty", "Unit Price", "Total Price 2020", "Average Annual Price"],
+              ["Almonds", 2, 8.0, "=C3 * D3", "=Sample2019Data!E3 + E3 / 2"],
+              ["Coffee", 1, 36.5, "=C4 * D4", "=Sample2019Data!E4 + E4 / 2"],
+              ["Chocolate", 5, 11.2, "=C5 * D5", "=Sample2019Data!E5 + E5 / 2"]
+            ];
+
+            // Style the tables.
+            const range2 = sheet2.getRange("B2:F5");
+            range2.values = data2;
+            range2.format.autofitColumns();
+
+            const header1 = range1.getRow(0);
+            header1.format.fill.color = "#4472C4";
+            header1.format.font.color = "white";
+
+            const header2 = range2.getRow(0);
+            header2.format.fill.color = "#4472C4";
+            header2.format.font.color = "white";
+
+            sheet1.activate();
+            sheet2.activate();
+
+            await context.sync();
+          });
+        }
+
+        /* Select a cell with direct precedents on one worksheet. */
+        async function selectFirstCell() {
+          await Excel.run(async (context) => {
+            // Ensure correct worksheet is active.
+            const sheet = context.workbook.worksheets.getItem("Sample2020Data");
+            sheet.activate();
+            sheet.load("Sample2020Data");
+
+            // Select the cell.
+            const range = sheet.getRange("E4");
+            range.select();
+            await context.sync();
+          });
+        }
+
+        /* Select a cell with direct precedents on both worksheets. */
+        async function selectSecondCell() {
+          await Excel.run(async (context) => {
+            // Ensure correct worksheet is active.
+            const sheet = context.workbook.worksheets.getItem("Sample2020Data");
+            sheet.activate();
+            sheet.load("Sample2020Data");
+
+            // Select the cell.
+            const range = sheet.getRange("F5");
+            range.select();
+            await context.sync();
+          });
+        }
+
+        /* Use the direct precedents API to detect precedent cells. */
+        async function getDirectPrecedents() {
+          await Excel.run(async (context) => {
+            let range = context.workbook.getActiveCell();
+            var rac = range.getDirectPrecedents();
```

Comment:
Can we rename `rac` to something more informative? Maybe `directPrecedents`? Also, please change `var` to `let` as per TypeScript guidelines.

Reply:
Sure! This block is directly from the spec code sample. Can you point me towards a good TypeScript guidelines reference?

Reply:
Here's a thorough description of `var`, `let`, and `const`: https://www.typescriptlang.org/docs/handbook/variable-declarations.html

Basically, there's some small scoping problems a user can encounter with `var` that are avoided by using `let`. Since we don't know how people are going to copy-paste our code, we want to try and make the most stable blocks we can.